### PR TITLE
upgrade to phpunit 11.1

### PR DESCRIPTION
**Pull request by mjhmatt**
#### What?

Upgrading php unit 11.1 so we have support for php 8.2 and can upgrade laravel to 11

#### Tickets / Documentation
https://bigcommercecloud.atlassian.net/browse/PHPMNT-88

### CODE REVIEW

**Review by bc-vincent-zhao — APPROVED** [review]

**Comment by TomA-R** [comment]
Would be good to also address the deprecations:

```
There were 8 PHPUnit test runner deprecations:

1) Metadata found in doc-comment for class Tests\Dummy\DummyTest. Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.

2) Metadata found in doc-comment for method Tests\MockInjectorTest::testCreate(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.

3) Metadata found in doc-comment for class Tests\MockInjectorTest. Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.

4) Metadata found in doc-comment for method Tests\MockInjectorTest::testInvoke(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.

5) Metadata found in doc-comment for method Tests\MockInjectorTest::testCheckPredictions(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.

6) Metadata found in doc-comment for method Tests\MockInjectorTest::testGetAllMocks(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.

7) Metadata found in doc-comment for method Tests\MockInjectorTest::testGetMock(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.

8) Metadata found in doc-comment for method Tests\MockInjectorTest::testWithoutDependencies(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
```

**Comment by mjhmatt** [comment]
> Would be good to also address the deprecations:
> 
> ```
> There were 8 PHPUnit test runner deprecations:
> 
> 1) Metadata found in doc-comment for class Tests\Dummy\DummyTest. Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> 
> 2) Metadata found in doc-comment for method Tests\MockInjectorTest::testCreate(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> 
> 3) Metadata found in doc-comment for class Tests\MockInjectorTest. Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> 
> 4) Metadata found in doc-comment for method Tests\MockInjectorTest::testInvoke(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> 
> 5) Metadata found in doc-comment for method Tests\MockInjectorTest::testCheckPredictions(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> 
> 6) Metadata found in doc-comment for method Tests\MockInjectorTest::testGetAllMocks(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> 
> 7) Metadata found in doc-comment for method Tests\MockInjectorTest::testGetMock(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> 
> 8) Metadata found in doc-comment for method Tests\MockInjectorTest::testWithoutDependencies(). Metadata in doc-comments is deprecated and will no longer be supported in PHPUnit 12. Update your test code to use attributes instead.
> ```

```
PHPUnit 11.2.7 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.9
Configuration: /Users/matt.hill/bigcommerce/mock-injector/phpunit.xml

........                                                            8 / 8 (100%)

Time: 00:00.049, Memory: 10.00 MB

OK (8 tests, 14 assertions)
```

**Review by funivan — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -22,10 +22,10 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require": {
-        "php": "^8.1",
+        "php": "^8.2",
         "bigcommerce/injector": "^4.1",
         "phpspec/prophecy-phpunit": "^2.0",
-        "phpunit/phpunit": "^9.5 || ^10.0"
+        "phpunit/phpunit": "^9.5 || ^11.0"
```
Comment (filakhtov) [inline composer.json:28]: Any reason to not also keep `^10.0`?

File: tests/MockInjectorTest.php [hunk 2]
```diff
@@ -9,10 +9,16 @@
...
-/**
- * @coversDefaultClass \Bigcommerce\MockInjector\MockInjector
- */
+#[CoversClass(MockInjector::class)]
+#[CoversMethod(MockInjector::class, 'create')]
+#[CoversMethod(MockInjector::class, 'invoke')]
+#[CoversMethod(MockInjector::class, 'checkPredictions')]
+#[CoversMethod(MockInjector::class, 'getAllMocks')]
+#[CoversMethod(MockInjector::class, 'getMock')]
+#[CoversMethod(MockInjector::class, '__construct')]
```
Comment (filakhtov) [inline tests/MockInjectorTest.php:21]: Any reason why you've opted to push `CoversMethod` from individual tests to the top level annotation? This changes the behaviour slightly, where `testCreate()` for example will contribute to the coverage of `__construct()` without actually performing any assertions on the outcome.

**Review by filakhtov — APPROVED** [review]

File: tests/MockInjectorTest.php [hunk 3]
```diff
@@ -9,10 +9,9 @@
 use Prophecy\Prophecy\ObjectProphecy;
 use Tests\Dummy\DummyDependency;
 use Tests\Dummy\DummySubDependency;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\CoversMethod;
```
[the remark this answers is not in the document]
Comment (mjhmatt) [reply tests/MockInjectorTest.php]: Done

**Review by filakhtov — APPROVED** [review]

**Pull request merged**